plan/physical_plan_builder.go

```diff
@@ -855,6 +866,8 @@ func (p *Selection) convert2PhysicalPlanEnforce(prop *requiredProperty) (*physic
 	if prop.limit != nil && len(prop.props) > 0 {
 		if t, ok := info.p.(physicalDistSQLPlan); ok {
 			t.addTopN(p.ctx, prop)
+		} else if _, ok := info.p.(*Selection); !ok {
```

Comment:
Oh, I remembered, the child has pulled the parent Selection down.
This design is anit-pattern.